### PR TITLE
DAOS-2791 docs: Document DAOS Agent in Admin Guide

### PR DESCRIPTION
**Pull request by dpquigl**
Added text to the quickstart and admin deployment guides pertaining to running
and configuring daos_agent. The deplyment guide adds a whole new section about
the agent, certificate usage, and fully describing the agent config file
format.

### CODE REVIEW

File: doc/admin/deployment.md [hunk 1]
```diff
@@ -1106,10 +1114,128 @@ Replace 16G with the desired tmpfs size.
...
+
+log\_file: /tmp/daos\_agent2.log
+
+## Agent Startup
+--------------
+
+DAOS Agent is a standalone application to be run on each compute node. It can be configured to use secure communications (default) or can be allowed to communicate with the control plane over unencrypted channels. The example below for executing daos_agent specifies to operate in insecure mode as certificate support is not fully integrated into DAOS as of 0.6
+
+For further details on building and running DAOS see the
+[Quickstart guide](../quickstart.md).
```
Comment (johannlombardi) [inline doc/admin/deployment.md]: I am about the remove the quickstart guide, so please insert instructions there.

File: doc/admin/deployment.md [hunk 2]
```diff
@@ -1106,10 +1114,128 @@ Replace 16G with the desired tmpfs size.
...
+development and will be delivered in DAOS v1.0. Initial support for certificates has been added to DAOS and can be disable either via the command line or in the DAOS Agent configuration file.
+
+### Server Configuration File
+
+The `daos_agent` configuration file is parsed when starting the
+`daos_agent` process. The configuration file location can be specified
+on the command line (`daos_agent -h` for usage) or default location
+(`install/etc/daos_agent.yml`).
+
+Parameter descriptions are specified in [daos_agent.yml](../../utils/config/daos_agent.yml)
```
Comment (johannlombardi) [inline doc/admin/deployment.md]: Could just use absolute path /utils/config/daos_agent.yml

File: doc/admin/deployment.md [hunk 3]
```diff
@@ -330,7 +330,7 @@ nodes before starting it.
 
 The DAOS security framework relies on certificates to authenticate
 administrators. The security infrastructure is currently under
-development and will be delivered in DAOS v1.0.
+development and will be delivered in DAOS v1.0. Initial support for certificates has been added to DAOS and can be disable either via the command line or in the DAOS server configuration file.
```
Comment (kjacque) [inline doc/admin/deployment.md]: Minor typo: "can be disable" -> "can be disabled"

File: doc/admin/deployment.md [hunk 4]
```diff
@@ -810,7 +818,7 @@ man page for additional options.
 To start the DAOS server, run:
 
 orterun -np &lt;num\_servers&gt; --hostfile \${hostfile}
---enable-recovery --report-uri \${urifile} daos\_server
+--enable-recovery --report-uri \${urifile} daos\_server -i
```
Comment (kjacque) [inline doc/admin/deployment.md]: Should probably note what "-i" does above before asking folks to use it. This overrides the allow_insecure line in the config file, right?

File: doc/admin/deployment.md [hunk 5]
```diff
@@ -1106,16 +1114,131 @@ Replace 16G with the desired tmpfs size.
...
 Agent Configuration
 -------------------
+This section addresses how to configure the DAOS servers on the storage
+nodes before starting it.
+
+### Certificate Generation
+
+The DAOS security framework relies on certificates to authenticate
+administrators. The security infrastructure is currently under
+development and will be delivered in DAOS v1.0. Initial support for certificates has been added to DAOS and can be disable either via the command line or in the DAOS Agent configuration file.
```
Comment (kjacque) [inline doc/admin/deployment.md]: sp: "can be disabled"

Additionally, just a nit, but the line length should probably be broken up to match up to the others in this paragraph.

File: doc/admin/deployment.md [hunk 6]
```diff
@@ -1106,16 +1114,131 @@ Replace 16G with the desired tmpfs size.
...
+
+#### Configuration File Options
+
+This section lists the default empty configuration listing all the
+options (living documentation of the config file). Live examples are
+available at
+<https://github.com/daos-stack/daos/tree/master/utils/config>
+
+The Location of this configuration file is determined by first checking
+for the path specified through the -o option of the daos\_agentcommand
```
Comment (kjacque) [inline doc/admin/deployment.md]: Missing space between "daos_agent" and "command"

File: doc/admin/deployment.md [hunk 7]
```diff
@@ -1106,16 +1114,131 @@ Replace 16G with the desired tmpfs size.
...
+log\_file: /tmp/daos\_agent2.log
+
+## Agent Startup
+--------------
+
+DAOS Agent is a standalone application to be run on each compute node. It can be configured to use secure communications (default) or can be allowed to communicate with the control plane over unencrypted channels. The example below for executing daos_agent specifies to operate in insecure mode as certificate support is not fully integrated into DAOS as of 0.6
+
+To start the DAOS Agent, run:
+```
+daos_agent -i
```
Comment (kjacque) [inline doc/admin/deployment.md]: Same comment about the "-i" param as above.

File: doc/admin/deployment.md [hunk 8]
```diff
@@ -1106,16 +1114,131 @@ Replace 16G with the desired tmpfs size.
...
+```
 
 System Validation
 -----------------
 
 To validate that the DAOS system is properly installed, the daos\_test
-suite can be executed:
+suite can be executed. Ensure the DAOS Agent is configuerd and running before running daos\_test:
```
Comment (kjacque) [inline doc/admin/deployment.md]: sp: "configured"

**Comment by johannlombardi** [comment]
Let's land this patch since we really need to have this documented. Please provide a separate PR to address Kris' comments. Thanks in advance.

**Pull request merged**